query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala

```diff
@@ -83,17 +81,12 @@ final case class LabelValuesDistConcatExec(id: String,
                         queryConfig: QueryConfig): Observable[RangeVector] = {
     qLogger.debug(s"NonLeafMetadataExecPlan: Concatenating results")
     val taskOfResults = childResponses.map {
-      case (QueryResult(_, _, result), _) => result
+      case (QueryResult(_, _, result), _) => result.map(_.asInstanceOf[SerializableRangeVector])
       case (QueryError(_, ex), _)         => throw ex
     }.toListL.map { resp =>
-      var metadataResult = scala.collection.mutable.Set.empty[Map[ZeroCopyUTF8String, ZeroCopyUTF8String]]
-      resp.foreach(rv => {
-        metadataResult ++= rv(0).rows.map(rowReader => {
-          val binaryRowReader = rowReader.asInstanceOf[BinaryRecordRowReader]
-          rv(0).schema.toStringPairs(binaryRowReader.recordBase, binaryRowReader.recordOffset)
-            .map(pair => pair._1.utf8 -> pair._2.utf8).toMap
-        })
-      })
+      val metadataResult = resp.map { rv =>
+        rv(0).key.labelValues.map(pair => pair._1 -> pair._2)
```

Comment:
@vishramachandran this may not work, as the key is a dummy key. Map of label name/values are returned as rows, schema has one column with type as Map.

Reply:
Ah - I think I get it now.

Reply:
Fixed.